### PR TITLE
Add undefined to Symbol.declarations' type

### PR DESCRIPTION
**Pull request by sandersn**
Symbol.declarations now has type `Declaration[] | undefined`.

### CODE REVIEW

**Comment by ajafff** [comment]
While you are at it, could you fix the rest of #24706 as well?

**Review by andrewbranch — APPROVED** [review]

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -1241,7 +1241,7 @@ namespace ts {
                 // as we will already report a "Declaration name conflicts..." error, and this error
                 // won't make much sense.
                 if (target !== globalThisSymbol) {
-                    error(getNameOfDeclaration(source.declarations[0]), Diagnostics.Cannot_augment_module_0_with_value_exports_because_it_resolves_to_a_non_module_entity, symbolToString(target));
+                    error(source.declarations && getNameOfDeclaration(source.declarations[0]), Diagnostics.Cannot_augment_module_0_with_value_exports_because_it_resolves_to_a_non_module_entity, symbolToString(target));
```
Comment (elibarzilay) [inline src/compiler/checker.ts:1244]: Newlines between the arguments?

**Review by elibarzilay — APPROVED** [review]
No good understanding of the highlevel change, but two picky comments...

**Comment by sandersn** [comment]
@ajaff I'm going to see how easy it is to fix `valueDeclaration` too.

**Pull request merged**